### ISSUE TITLE
Style.repeat with Style.delay jerkiness

### ISSUE THREAD

**Issue by jesperp** [mdgriffith/elm-style-animation#11]
First of all, thanks for a great library!
I'm having some issues with my repeated animations that has delays. Not sure if this is even supported...

```
  Style.repeat 4
    |> Style.spring { stiffness =  300, damping = 18 }
    |> Style.to [ Left 150 Px , Opacity 1 ]
    |> Style.andThen
    |> Style.delay (0.5 * Time.second)
    |> Style.to [ Left 0 Px , Opacity 0.3 ]
    |> Style.on model.style
```

Here's how it looks:
![jerkiness](https://cloud.githubusercontent.com/assets/376921/16900331/85352902-4c22-11e6-82f5-3eb194203027.gif)

Jerkiness also seem to sync with my delay (0.5sec).

Any ideas?

[Full source](https://gist.github.com/jesperp/16db784aa849393ca54aa2e7d482bf9b)

**Comment by mdgriffith** [comment]
Thanks!  and thank you for providing a minimal working example.  Hmm, interesting.  It should be supported, this is definitely a bug.  It goes away if you manually compose the animation:

```
              style =
                    (List.foldl
                        (\i anim ->
                            let
                                animation =
                                    if i == 0 then
                                        anim
                                    else
                                        anim |> Style.andThen
                            in
                                animation
                                    |> Style.spring { stiffness = 300, damping = 18 }
                                    |> Style.to [ Left 150 Px, Opacity 1 ]
                                    |> Style.andThen
                                    |> Style.spring { stiffness = 300, damping = 18 }
                                    |> Style.delay (0.5 * Time.second)
                                    |> Style.to [ Left 0 Px, Opacity 0.3 ]
                        )
                        Style.animate
                        [1..5]
                    )
                        |> Style.on model.style

```

So, this looks like something specific to how repeat and delay interact.  We'll see if I can track it down :)

**Comment by mdgriffith** [comment]
I believe this should be fixed in the latest (v3.0.0) release.
